Source/FulcrumFS.Videos/VideoProcessor.cs

```diff
@@ -0,0 +1,236 @@
+    /// Default is a list containing <see cref="MediaContainerFormat.MP4" />.
+    /// </summary>
+    public IReadOnlyList<MediaContainerFormat>? ResultMediaContainerFormat
+    {
+        get;
+        init
+        {
+            if (value is null)
+            {
+                field = null;
+                return;
+            }
+
+            IReadOnlyList<MediaContainerFormat> result = [.. value];
+
+            if (result.Count is 0)
+                throw new ArgumentException("Formats cannot be empty.", nameof(value));
+
+            if (result.Any((x) => x is null))
+                throw new ArgumentException("Formats cannot contain null values.", nameof(value));
+
+            if (result.Distinct().Count() != result.Count)
+                throw new ArgumentException("Formats cannot contain duplicates.", nameof(value));
+
+            if (!result[0].SupportsWriting)
+                throw new ArgumentException("The first format in the list must support writing.", nameof(value));
+
+            field = result;
+        }
+    } = [MediaContainerFormat.MP4];
+
+    /// <summary>
+    /// Gets or initializes a value indicating whether to strip metadata from this stream.
+    /// If set to <see langword="null" />, metadata preservation is undefined, and it may only be partially preserved.
+    /// Default is <see langword="false" />.
+    /// </summary>
+    public bool? StripMetadata { get; init; } = false;
+
+    /// <summary>
+    /// Gets or initializes a value indicating whether to ensure the 'moov atom' in an MP4 file is at the beginning of the file.
+    /// Note: this does not enable true streaming, but it does allow playback to begin before the entire streams are downloaded.
+    /// Default is <see langword="true" />.
+    /// </summary>
+#pragma warning disable SA1623 // Property summary documentation should match accessors
+    public bool ForceProgressiveDownload { get; init; } = true;
+#pragma warning restore SA1623 // Property summary documentation should match accessors
+
+    /// <summary>
+    /// Gets or initializes a value indicating whether to preserve unrecognized streams in the output video.
+    /// Note: <see langword="null" /> is the default behavior, meaning unrecognized streams are removed when changing container format only - however more
+    /// streams (e.g., subtitle streams) may become recognized in the future.
+    /// Default is <see langword="null" />.
+    /// </summary>
+    public bool? PreserveUnrecognizedStreams { get; init; }
+
+    /// <summary>
+    /// Gets or initializes a value indicating whether to trim the thumbnail from the video.
+    /// Default is <see langword="true" />.
+    /// Note: disabling trimming may not be respected if the container format is changed.
+    /// </summary>
+#pragma warning disable SA1623 // Property summary documentation should match accessors
+    public bool StripThumbnail { get; init; } = true;

```

Comment:
I think we can align this with how `Images` works, i.e have 3 options on a `StripVideoMetadataMode` enum.